### PR TITLE
Comment on Diff feature.

### LINKED ISSUE #3: Commenting the diff with individual comments
https://github.com/jenkinsci/violation-comments-to-gitlab-plugin/issues/1

### PR DESCRIPTION
**Pull request by bonzani**
Implement the comment on Diff feature for GitLab. This PR depends on timols/java-gitlab-api#318

Closes #3, closes jenkinsci/violation-comments-to-gitlab-plugin#1

### CODE REVIEW

File: src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java [hunk 1]
```diff
@@ -20,184 +20,192 @@
...
+		return !commentOnlyChangedContent || lineFoundOpt.isPresent();
+	}
+
+	@Override
+	public boolean shouldCreateBulkComment() {
+		return violationCommentsToGitLabApi.getCreateCommentWithAllSingleFileComments();
+	}
+
+	@Override
+	public boolean shouldCommentOnTheDiff() {
```
Comment (tomasbjerre) [inline src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java]: You probably did not mean to hard code this to false.

**Comment by tomasbjerre** [comment]
I'm not sure what this erroprone failure is about in Travis...

I'll try to have a closer look at this as soon as I can!

File: src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java (hunk 1 above)
Comment (bonzani) [reply src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java]: Yes you are right. I still need to add a config parameter for that one. I will do it as soon as possible.

**Comment by bonzani** [comment]
Thank you. It seems that it was cancelled, but I cannot see by what.

**Comment by tomasbjerre** [comment]
It was a new version of the Errorprone plugin. Fixed in master branch now so just rebase on that.

File: src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java [hunk 2]
```diff
@@ -20,184 +20,192 @@
...
+	}
+
+	@Override
+	public boolean shouldCreateCommentWithAllSingleFileComments() {
+		return violationCommentsToGitLabApi.getCreateCommentWithAllSingleFileComments();
+	}
+
+	@Override
+	public boolean shouldCreateSingleFileComment() {
+		return violationCommentsToGitLabApi.getCreateCommentPerViolation();
```
Comment (tomasbjerre) [inline src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java]: Should be:
```
  @Override
  public boolean shouldCreateSingleFileComment() {
    return violationCommentsToGitHubApi.getCreateSingleFileComments();
  }
```

**Comment by tomasbjerre** [comment]
I merged the code into a feature branch within this repo. And opened a new PR from that feature branch: https://github.com/tomasbjerre/violation-comments-to-gitlab-lib/pull/8

I needed to make some adjustments to make it compile.

**Pull request closed without merging**